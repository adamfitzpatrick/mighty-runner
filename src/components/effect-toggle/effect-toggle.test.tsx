import { h } from 'preact'
import deep, { RenderContext } from 'preact-render-spy'
import EffectToggle from '.'
import { Effect } from '@models'


describe('EffectToggle component', () => {
  let effect: Effect
  let toggleHandler: (id: string) => void
  type Props = {
    effect: Effect
    toggleHandler: typeof toggleHandler
  }
  let effectToggle: RenderContext<Props, {}>

  beforeEach(() => {
    effect = {
      id: '1',
      name: 'effect',
      target: [ 'target' ],
      active: false,
      value: 1
    }
    toggleHandler = jest.fn()
    effectToggle = deep(<EffectToggle effect={effect} toggleHandler={toggleHandler} />)
  })

  test('should render correctly', () => {
    expect(effectToggle).toMatchSnapshot()
  })

  test('when clicked should call the toggle handler', () => {
    effectToggle.find('input').simulate('click')
    expect(toggleHandler).toHaveBeenCalledWith('1')
  })
})
