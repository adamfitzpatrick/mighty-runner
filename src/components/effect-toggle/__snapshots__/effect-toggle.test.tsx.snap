// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EffectToggle component should render correctly 1`] = `
preact-render-spy (1 nodes)
-------
<label
  class="effectToggle"
  aria-label="effect, value 1, inactive"
  aria-live={true}
>
  <input
    type="checkbox"
    value="1"
    onClick={[Function onClick]}
    checked={false}
    class="effectInput"
   />
  <span class="effectFocusIndicator"></span>
  <span class="effectIndicator"></span>
  <span class="effectName">effect</span>
  <span class="effectValue">1</span>
</label>

`;
